libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -93,6 +101,41 @@ class AudioSource(TypedDict):
     type: str
 
 
+class FileSource(TypedDict):
+    src: str
+    type: str
+
+
+def create_csv_file(
+    dataset: str,
+    config: str,
+    split: str,
+    data: Any,
+    headers: List[str],
+    assets_base_url: str,
+    file_name: str,
+    assets_directory: StrPath,
+) -> FileSource:
+    dir_path, url_dir_path = create_file_dir(
+        dataset=dataset,
+        config=config,
+        split=split,
+        assets_directory=assets_directory,
+    )
+    makedirs(dir_path, ASSET_DIR_MODE, exist_ok=True)
+    file_path = dir_path / file_name
+
+    with open(file_path, "w") as output_file:
+        fc = csv.DictWriter(
+            output_file,
+            fieldnames=headers,
+        )
+        fc.writeheader()
+        fc.writerows(data)
```

Comment:
we rely on the fact that the caller of this function passes coherent values for `headers` and `data`:
- same set of values
- in the same order

Is it OK? maybe add some docstring for that.

Reply:
Also: we never created unit tests for the assets functions. Should we use this PR to add them?

Reply:
Sure, I will add them